src/infrastructure/content/ContentFactory.java

```diff
@@ -0,0 +1,9 @@
+package content;
+
+public final class ContentFactory {
+	
+    public static IContentCommunicator getContentCommunicator() {
+        IContentCommunicator communicator = new ContentCommunicator();
+        return communicator;
+    }
+}
```

Comment:
Refer [this](https://stackoverflow.com/a/7704020) for a singleton pattern implementation. The only difference is that we want the ContentCommunicator to have only one instance instead of the factory. So change that private static variable to ContentCommunicator and in the getContentCommunicator() function, it will be instance = new ContentCommunicator() instead of factory. Static variables and methods can be accessed without instantiating the Factory class so make sure to make the variable and the method static. The constructor should be private as the factory is not initialised normally